src/k8s-engine/client.js

```diff
@@ -0,0 +1,147 @@
+'use strict';
+
+// This file contains wrappers to interact with the installed Kubernetes cluster
+
+const https = require("https");
+const net = require("net");
+const k8s = require("@kubernetes/client-node");
+
+/**
+ * KubeClient is a Kubernetes client that will _only_ manage the cluster we spin
+ * up internally.  The user should call initialize() once the cluster has been
+ * created.
+ */
+class KubeClient {
+    #kubeconfig = new k8s.KubeConfig();
+    /**
+     * @type k8s.PortForward?
+     */
+    #forwarder = null;
+    /**
+     * @type net.Server?
+     */
+    #server = null;
+
+    /**
+     * initialize the KubeClient so that we are ready to talk to it.
+     */
+    initialize() {
+        this.#server?.close();
+        this.#server = null;
+        this.#kubeconfig.loadFromDefault();
+        this.#kubeconfig.currentContext = 'rancher-desktop';
+        this.#forwarder = new k8s.PortForward(this.#kubeconfig, true);
+    }
+
+    /**
+     * @type k8s.CoreV1Api
+     */
+    get #coreV1API() {
+        return this.#_coreV1API ||= this.#kubeconfig.makeApiClient(k8s.CoreV1Api);
+    }
+    #_coreV1API = null;
+
+    /**
+     * Return a pod that is part of a given endpoint and ready to receive traffic.
+     * @param {string} namespace The namespace in which to look for resources.
+     * @param {string} endpointName the name of an endpoint that controls ready pods.
+     * @returns {Promise<k8s.V1Pod>}
+     */
+    async getActivePod(namespace, endpointName) {
+        console.log(`Attempting to locate ${endpointName} pod...`);
+        // Loop fetching endpoints, until it matches at least one pod.
+        /** @type k8s.V1ObjectReference? */
+        let target = null;
+        // TODO: switch this to using watch.
+        for (; ;) {
+            /** @type k8s.V1EndpointsList */
+            let endpoints;
+            ({ body: endpoints } = await this.#coreV1API.listNamespacedEndpoints(namespace, { headers: { name: endpointName } }));
+            console.log(`Got ${endpointName} endpoints: ${endpoints}`);
+            target = endpoints?.items?.pop()?.subsets?.pop()?.addresses?.pop()?.targetRef;
+            console.log(`Got ${endpointName} target: ${target?.namespace}:${target?.name}`);
```

Comment:
These are causing repeated log entries of ...

```
Got homestead endpoints: [object Object]
Got homestead target: undefined:undefined
```

The `[object Object]` should be fixed. At what point does this not get written to the log on all the loops? Should we add a debug mode for that?